q2.java

```diff
@@ -0,0 +1,113 @@
+
+
+//Implementing a linked list
+class LinkedList<T>{
+	Node<T> head;
+	
+	public LinkedList(){
+		head = null;
+	}
+	
+	public void appendLast(T data){
+		Node<T> last = new Node<T>(data);
+		Node<T> current = this.head;
+		if(current == null){
+			head = last;
+			return;
+		}
+		while(current.next!=null){
+			current = current.next;
+		}
+		current.next=last;		
+	}
+	
+	/** 
+	 * The idea is to create a window of length k where
+	 * right margin is pointed by current
+	 * left margin is pointed by follower
+	 * when current reaches end of linked linked, follower points to kth to last element

```

Comment:
It would be good practice to also note the error cases and failure cases here in the JavaDoc. for example noting that the input must be >=0 and that invalid inputs or inputs that exceed the list length will cause the method to return null.